File: 2021/Day 02/aoc02_1.cpp
#include <iostream>
#include <fstream>
using namespace std;

int main() {
    ifstream infile;
    infile.open("aoc02_input.txt");

    int x = 0;
    int y = 0;
    string d;
    int s;
    
    while(infile >> d) {
        infile >> s;

        if(d[0] == 'f') {
            x += s;
        } else if(d[0] == 'u') {
            y -= s;
        } else if(d[0] == 'd') {
            y += s;
        }
    }
    cout << x * y;

    return 0;
}

File: 2021/Day 02/aoc02_1.py
file = open("aoc02_input.txt")
l = file.read().split("\n")
file.close()

x = 0
y = 0
for i in l:
    d, s = i.split()
    s = int(s)

    if d[0] == "f":
        x += s
    
    if d[0] == "u":
        y -= s
    
    if d[0] == "d":
        y += s

print(x * y)